bundles/event_admin/event_admin/src/celix_event_admin.c

```diff
@@ -61,15 +64,24 @@ typedef struct celix_event_entry {
     celix_long_hash_map_t* eventHandlers;//key: event handler service id, value: null
 }celix_event_entry_t;
 
+typedef struct celix_event_seq_id_cache {
+    struct timespec lastModified;
+    long seqIdBuffer[CELIX_EVENT_ADMIN_MAX_EVENT_SEQ_ID_CACHE_SIZE];
+}celix_event_seq_id_cache_t;
+
 struct celix_event_admin {
     celix_bundle_context_t* ctx;
     celix_log_helper_t* logHelper;
     unsigned int handlerThreadNr;
-    celix_thread_rwlock_t lock;//projects: channels,eventHandlers
+    const char* fwUUID;
+    celix_thread_rwlock_t lock;//projects: channels,eventHandlers,eventSeqIdCache,remoteProviderServices
```

Comment:
and nextSeqId

Reply:
The `nextSeqId` is no longer protected by `lock` and is instead an atomic operation.